### PR TITLE
[DOC] add high_res demo

### LINKED ISSUE #432: [BUG]:  Selecting an `anat` image from another session does not works
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Expected Behavior

Being able to use an `anat` image from eg `ses-008` for a task in `ses-006` in `bidsSpatialPrepro(opt);`

### Current Behavior

`bidsSpatialPrepro(opt);` does not run for a task in eg session 006 if in the `opt` is selected 

```
 opt.anatReference.type = 'T1w';
 opt.anatReference.session = '008';
```
`getAnatFilename(BIDS, subLabel, opt)` looks for an `anat` image with the same session label of the task , line 80 report the task session ( 006) and later it checks if it is the same of the provided (008) and being different it throws and error.

Below the derivatives folder content

```
.
├── JOBS
│   └── mtMstLocalizerNoStatic
│       └── sub-pilot001

│           ├── 20211112_2218_jobs_matlabbatch_SPM12_STC.json
│           └── 20211112_2218_jobs_matlabbatch_SPM12_STC.mat
├── dataset_description.json
├── participants.json
├── participants.tsv
└── sub-pilot001
    ├── ses-006
    │   ├── func
    │   │   ├── asub-pilot001_ses-006_task-mtMstLocalizerNoStatic_run-001_bold.nii
    │   │   ├── sub-pilot001_ses-006_task-mtMstLocalizerNoStatic_run-001_bold.json
    │   │   ├── sub-pilot001_ses-006_task-mtMstLocalizerNoStatic_run-001_bold.nii
    │   │   └── sub-pilot001_ses-006_task-mtMstLocalizerNoStatic_run-001_events.tsv
    │   ├── sub-pilot001_ses-006_scans.json
    │   └── sub-pilot001_ses-006_scans.tsv
    └── ses-008
        └── anat
            ├── sub-pilot001_ses-008_T1w.json
            └── sub-pilot001_ses-008_T1w.nii
```

### Error message

```
Error using getAnatFilename>getAnatError (line 109)
Requested session 008 for anatomical unavailable for subject pilot001. All available sessions listed
above.

Error in getAnatFilename>checkAvailableSessions (line 94)
      getAnatError(msgID, msg);

Error in getAnatFilename (line 28)
  anatSession = checkAvailableSessions(BIDS, subLabel, opt, anatSession);

Error in setBatchSelectAnat (line 32)
  [anatImage, anatDataDir] = getAnatFilename(BIDS, subLabel, opt);

Error in bidsSpatialPrepro (line 58)
    matlabbatch = setBatchSelectAnat(matlabbatch, BIDS, opt, subLabel);
```

### Environment

```markdown
- OS: macOS
- Matlab: 2017b
- SPM: SPM12 7487
- CPP_SPM: 1.1.3
```

### Anything else?

thansk <3

### PR DESCRIPTION
**Pull request by Remi-Gau**

### CODE REVIEW

File: initCppSpm.m [hunk 1]
```diff
@@ -2,6 +2,11 @@ function initCppSpm()
   %
   % adds the relevant folders to the path
   %
+  % To force reinitialization use ``clearvars -GLOBAL`` first
```
Comment (Remi-Gau) [inline initCppSpm.m:5]: not needed anymore since there is a `uninitCppSpm`

**Comment by Remi-Gau** [comment]
@marcobarilari 

i am going to break this PR into several smaller ones because there are too many conflicts in there.

**Comment by Remi-Gau** [comment]
I am most likely only leave here the high res demo stuff and not all the things that change CPP_SPM itself.

**Comment by marcobarilari** [comment]
Hey! great but keep in mind that we also have https://github.com/cpp-lln-lab/CPP_High-Res_fMRI where I developed further (not much) the high res stuff. I think separate it from cpp-spm and make it and adds-on of datalad fmri template might work better (also because many spm steps will be droped)

**Comment by Remi-Gau** [comment]
yeah sure no problem: worst case we just close this PR completely but I just did not want to drop some of the things we did here.

**Comment by marcobarilari** [comment]
the code has been moved in [cpp-lln-lab/CPP_HighRes_fMRI](https://github.com/cpp-lln-lab/CPP_HighRes_fMRI)

**Pull request closed without merging**